File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        int maxProd = INT_MIN;
        int leftProd=0;
        int rightProd=0; 
        int i =0;
        while(i<n){
            
            if(leftProd ==0) leftProd =1;
            if(rightProd ==0) rightProd =1;
            leftProd = leftProd * nums[i];
            rightProd= rightProd * nums[n-i-1];
            maxProd = max(maxProd , max(leftProd, rightProd));
            i++;
        }
        
        return int(maxProd);
    }
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:

        maxProd = -1e9
        leftProd = 1
        rightProd= 1
        n = len(nums)
        i=0
        for i in range(n):
            if leftProd == 0:
                leftProd = 1
            if rightProd == 0:
                rightProd =1
            leftProd*=nums[i]
            rightProd*=nums[n-i-1]

            maxProd = max(maxProd, max(leftProd, rightProd))
        return maxProd